### PR TITLE
Display absolute path on failed dependency resolve.

### PR DESCRIPTION
**Pull request by pomle**
When a file path can not be resolved the error message is cleaned
up but the relative path to file is displayed.

In cases where a mistake have been made this path can be very confusing,
like `Cannot resolve dependency './../../../../../../main.js'` for when
a path starts with `/`.

This PR resolves path into an absolute path in hope of
reducing confusion and minimize cognitive overhead of `../` chasing.

The above example would now display 
```
Cannot resolve dependency '/main.js' at '/main.js'
```
and for relative unresolvable path
```
Cannot resolve dependency './even-more/x.js' at '/mnt/c/Users/pom/Development/pom/parcel-implementation/src/script/more/even-more/x.js'
```

### CODE REVIEW

**Comment by devongovett** [comment]
The path that it currently logs is exactly the path that is in the code of the parent module. What do people think about changing this? On one hand it is nice that the error message tells you what in the code is incorrect, but you're right that relative dependencies like that can be hard to parse.

**Comment by pomle** [comment]
I thought this thru and it is possible this can be misleading due to
resolve function origin path may vary.

Let's put it on the back burner for now. I'll get back shortly.

**Comment by pomle** [comment]
@devongovett I looked into it a bit more and revised it. Was obviously a bit more complicated to conclude something, but my changes so far include resolving absolute path relative to current asset. Before it was working dir. 

Not sure it is the right thing to do for all use cases. The error message print system also seems to suffer from a bug where the message is clipped when spanning multiple lines. The clipping bug is really the one that bit me,

Principally, I believe exposing absolute search paths is crucial to easy debugging, but rewriting the error message might not be the correct way.

Let me know if you think this is worth spending more time on right now and I'll be happy to proceed.

**Comment by kingdaro** [comment]
I'm onboard with this change. Using absolute paths is also good for editors with hyperclick support in terminals and other places, where the path wouldn't have been resolvable the way it is now.

**Comment by pomle** [comment]
Test is failing due to an indeterministic test issue:
https://github.com/parcel-bundler/parcel/issues/219

Perhaps this ticket should be re-opened? Test errors still persist on c008bb0ac492dbac38e2f13017e3143f40359934

**Comment by devongovett** [comment]
Nice, best of both worlds. 👍

**Comment by devongovett** [comment]
Modified slightly in https://github.com/parcel-bundler/parcel/commit/bd1d6e806522fcb539a9a50b3cb6bfdf46ea2888 to only include the absolute path if the dependency was specified as a relative one. That way you don't get incorrect absolute paths for npm dependencies that aren't installed (e.g. `require('foo')`).
